### ISSUE TITLE
[Runtime Issue]: v1.5 macOS ARM release does not have the executable set on the Descent3 binary

### ISSUE THREAD

**Issue by Kreeblah** [DescentDevelopers/Descent3#532]
### Build Version

v1.5

### Operating System Environment

- [ ] Microsoft Windows (32-bit)
- [ ] Microsoft Windows (64-bit)
- [X] Mac OS X
- [ ] Linux (specify distribution and version below)

### CPU Environment

- [ ] x86 (32-bit Intel/AMD)
- [ ] x86\_64 (64-bit Intel/AMD)
- [ ] ARM (32-bit)
- [X] ARM64 (64-bit; sometimes called AArch64)
- [ ] Other (RISC V, PPC...)

### Game Modes Affected

- [X] Single player
- [X] Anarchy
- [X] Hyper-Anarchy
- [X] Robo-Anarchy
- [X] Team Anarchy
- [X] Capture the Flag
- [X] Bounty
- [X] Entropy
- [X] Hoard
- [X] Monsterball
- [X] Cooperative

### Description

I downloaded a copy of the macOS ARM release of v1.5, and noticed that it refused to run.  Trying to run the app bundle gave me an error in the GUI that said it couldn't be opened, and if I tried to open it in the terminal, I got this:

```
The application cannot be opened for an unexpected reason, error=Error Domain=RBSRequestErrorDomain Code=5 "Launch failed." UserInfo={NSLocalizedFailureReason=Launch failed., NSUnderlyingError=0x600002e78120 {Error Domain=NSPOSIXErrorDomain Code=111 "Unknown error: 111" UserInfo={NSLocalizedDescription=Launchd job spawn failed}}}
```

Looking into it, it seems that the Descent3 binary doesn't have the executable bit set, and instead has permissions set to 644 when extracted from the ZIP file.

**Comment by Lgt2x** [comment]
Hey, thanks for the report! Does running `chmod +x ./Descent3.app/Content/MacOS/Descent3` fix the problem?

**Comment by Kreeblah** [comment]
Oh, I should have mentioned this in my report, but it does, for the binary.  Running `Descent3` directly from the terminal works fine after setting the executable bit on it.  I did have to copy my game assets into the `MacOS` directory alongside the binary (which maybe isn't ideal as opposed to something like using a more standard location in `~/Library/Application Support/Descent3` or some such, but works for this), but it loads and runs fine at that point.

I hadn't actually tried the app bundle until now, though, but for some reason, it quits immediately, even after setting the executable bit on it.  I don't see what's wrong with the app bundle, though.  `CFBundleExecutable` in `Info.plist` seems to be set properly, things seem to be located correctly (the binary's in the right spot, the `Info.plist` file looks right, etc.).

**Comment by Lgt2x** [comment]
The way the game finds game data is very primitive currently, simply looking what's in at the current directory or `-setdir` argument if set, and exiting if it could not find it. This is less than ideal, but following the usage guide and using a terminal should be enough to get the game running. We plan to work on a proper installer with a signed MacOS bundle, and automatically looking for game data location for the next release to improve user experience.

I'll leave this issue open for now, there may be other players confused by this logic until we rework it

**Comment by tophyr** [comment]
> The way the game finds game data is very primitive currently, simply looking what's in at the current directory or -setdir argument if set

I am actively working on this specifically, and will add more options to set different subdirectories, FWIW. That won't really do much for the overall problem of "must manually put gamedata in appropriate locations" however.

**Comment by tophyr** [comment]
> but for some reason, it quits immediately, even after setting the executable bit on it

@Kreeblah, try running `xattr -c <path/to/Descent3.app>` (the overall `.app`, _not_ just the executable inside it). I had to do this in order to get around a security restriction preventing execution of unsigned apps, but I'm a bit of an odd case. Let us know if that fixes your problem - if it does, I'm not as odd as I thought and we'll need to update the `USAGE.md` file to include that step.

**Comment by Jayman2000** [comment]
> > The way the game finds game data is very primitive currently, simply looking what's in at the current directory or -setdir argument if set
> 
> I am actively working on this specifically, and will add more options to set different subdirectories, FWIW. That won't really do much for the overall problem of "must manually put gamedata in appropriate locations" however.

How does what you’re working on compare to #471?

**Comment by Kreeblah** [comment]
> > but for some reason, it quits immediately, even after setting the executable bit on it
> 
> @Kreeblah, try running `xattr -c <path/to/Descent3.app>` (the overall `.app`, _not_ just the executable inside it). I had to do this in order to get around a security restriction preventing execution of unsigned apps, but I'm a bit of an odd case. Let us know if that fixes your problem - if it does, I'm not as odd as I thought and we'll need to update the `USAGE.md` file to include that step.

Unfortunately, that still doesn't let me launch it from the app bundle.  I even tried it with `-r` to clear out the extended attributes recursively, just in case.  The symptoms I'm seeing there are different than what I've seen with security restrictions on app bundles in the past, though.  The icon quickly appears and disappears in the Dock.  If I run the app bundle from the terminal, there's nothing printed, either.

Is it possible that it might not be finding the game assets?  I don't recall offhand what the current working directory is when running an app bundle, and I don't know what the behavior is if the app bundle can't find the assets, but based on what happens in the terminal (it just prints an error saying it can't find a file and quits), that's what's making me wonder.  I did try copying all the game assets to the `Resources` directory in the app bundle in case that was the CWD, but, no success there, either.

**Comment by Lgt2x** [comment]
> Is it possible that it might not be finding the game assets? I don't recall offhand what the current working directory is when running an app bundle, and I don't know what the behavior is if the app bundle can't find the assets, but based on what happens in the terminal (it just prints an error saying it can't find a file and quits), that's what's making me wonder. I did try copying all the game assets to the `Resources` directory in the app bundle in case that was the CWD, but, no success there, either.

Yes, what's probably happening here is that the assets could not be found. The game closes without any output (which is a problem) My guess would be the CWD is the location of the `.app` bundle? I recall being able to run the game when .hog files and .app have the same parent folder. 

@tophyr @Jayman2000 @winterheart  I've opened #534 and #532 so we can discuss a long-term solution there. Please comment here to synchronize with each other. This will be a priority for 1.6

**Comment by halprin** [comment]
What should we do with this issue?  The original issue identified here is the executable bit not being set for the macOS build.  It is now set, and this is no longer an issue.

But this issue also starts talking about not finding the game assets.  That will be fixed, mostly, when #628 is merged.  I say "mostly" because [I remember](https://github.com/DescentDevelopers/Descent3/pull/628#issuecomment-2499234585) D3 working when moving the game assets to `~/Library/Application Support/Outrage Entertainment/Descent 3`, but I do not remember if they worked when keeping all the game assets next to `Descent3.app`.  #628 is old enough that the build artifacts have expired from that PR nor can I retrigger the build (maybe I'm missing something), so I can't retest.


_Anyway_, I guess what I'm asking is should we...
1. Close this issue as fixed because the original issue has been fixed for a while.
2. Keep this issue open for now and reconsider closing it when #628 is merged?

**Comment by Lgt2x** [comment]
Let's close this, the main issue is fixed. Game assets find logic is another problem, and we already have open issues and PRs regarding that
